### PR TITLE
Disable create/rename tag button if tag title exists or is empty

### PR DESCRIPTION
**Pull request by tobias-klein**
This introduces basic validation of the tag title when creating new tags and when renaming existing tags.

* The create/update tag button is disabled when the tag is empty or if it already exists.
* The button gets enabled again once it is not empty and does not exist yet.

### CODE REVIEW

File: app/frontend/controllers/tags_controller.js [hunk 1]
```diff
@@ -106,21 +106,54 @@ class TagsController {
...
         tags_controller.saveNewTag(event, "standard");
       }
+    }).on("keyup", async (event) => {
+      var tag_title = $('#new-standard-tag-title-input').val();
+      await this.updateButtonStateBasedOnTagTitleValidation(tag_title, 'create-tag-button');
     });
   }
 
+  async updateButtonStateBasedOnTagTitleValidation(tagTitle, buttonId) {
+    var tag_existing = await this.tagExists(tagTitle);
```
Comment (zhuiks) [inline app/frontend/controllers/tags_controller.js:139]: Would `tagExists` be fast enough for each `keyup` event?

**Review by zhuiks — COMMENTED** [review]
Everything looks fine.
I have one question and one comment

File: app/frontend/controllers/tags_controller.js (hunk 1 above)
Comment (tobias-klein) [reply app/frontend/controllers/tags_controller.js:139]: On the desktop - yes!
On Android - I still need to test.

Comment (tobias-klein) [reply app/frontend/controllers/tags_controller.js:139]: Sufficiently fast on Android as well - just tested.

**Pull request merged**